packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Rector\Php\Tests\Rector\FuncCall\CountOnNullRector\Fixture;
+
+use Rector\Php\Tests\Rector\FuncCall\CountOnNullRector\Source\CountableClass;
+
+final class LocalVariable
+{
+    public function methodWithLocalVariableInitializedAsArray(array $anArray): ?array
+    {
+        $results = [];
+
+        foreach ($anArray as $value) {
+            $results[] = $value;
+        }
+
+        return count($results) > 0 ? $results : null;
+    }
+
+    public function methodWithLocalVariableArrayAsResult(string $string, array $regexes): ?array
+    {
+        $results = [];
+
+        foreach ($regexes as $toFind) {
+            if (false !== preg_match_all($toFind, $string, $result)) {
+                $results[] = $result[0];
+            }
+        }
+
+        // Flatten the array
+        $results = array_merge(...$results);
+
+        return count($results) > 0 ? $results : null;
+    }
+
+    public function methodWithLocalVariableInitializedAsArrayAndTransformedIntoString(array $anArray): ?array
+    {
+        $results = [];
+
+        // This code may not have much sense,
+        // but in real world scenario it may
+        // appear very similar to this.
+        foreach ($anArray as $value) {
+            if ('condition' === $value) {
+                $results = 'transform into a string';
```

Comment:
TBH. I'm not sure this should be supported at all cause string is not countable in newer PHP versions, yet this probably will fail while checking code with phpstan.

Reply:
Yes, you are right. In my opinion, it is something that should be avoided at all.

I added it for completeness, but there's nothing against removing it.

ping @TomasVotruba